sdk/eventgrid/azure-eventgrid/azure/eventgrid/aio/_operations/_operations.py

```diff
@@ -89,21 +98,33 @@ async def _publish_cloud_event(  # pylint: disable=inconsistent-return-statement
         response = pipeline_response.http_response
 
         if response.status_code not in [200]:
+            if _stream:
+                await  response.read()  # Load the body in memory and close the socket
             map_error(status_code=response.status_code, response=response, error_map=error_map)
             raise HttpResponseError(response=response)
 
+        if _stream:
+            deserialized = response.iter_bytes()
+        else:
+            deserialized = _deserialize(
+                _models._models.PublishResult,  # pylint: disable=protected-access
+                response.json()
+            )
+
         if cls:
-            return cls(pipeline_response, None, {})
+            return cls(pipeline_response, deserialized, {}) # type: ignore
+
+        return deserialized # type: ignore
 
 
 
     @distributed_trace_async
-    async def _publish_cloud_events(  # pylint: disable=inconsistent-return-statements
+    async def _publish_cloud_events(  # pylint: disable=protected-access
         self,
         topic_name: str,
         events: List[_models._models.CloudEvent],
         **kwargs: Any
-    ) -> None:
+    ) -> _models._models.PublishResult:
```

Comment:
is the return type change here going to impact anything public facing ?

Reply:
Nope this function is private:)